src/fiatconnect/ReviewScreen.tsx

```diff
@@ -0,0 +1,340 @@
+        <PaymentMethod
+          provider={provider}
+          fiatAccount={fiatAccount}
+          fiatAccountSchema={fiatAccountSchema}
+          fiatAccountLogo={fiatAccountLogo}
+        />
+      </View>
+      <Button
+        style={styles.submitBtn}
+        type={BtnTypes.PRIMARY}
+        size={BtnSizes.FULL}
+        text={
+          flow === CICOFlow.CashIn
+            ? t('fiatConnectReviewScreen.cashIn.button')
+            : t('fiatConnectReviewScreen.cashOut.button')
+        }
+        onPress={() => {
+          Logger.debug(TAG, 'Button clicked')
+
+          // TODO(any): submit the transfer
+        }}
+      />
+    </SafeAreaView>
+  )
+}
+
+function Amount({
+  flow,
+  cicoQuote,
+  tokenAddress,
+}: {
+  flow: CICOFlow
+  cicoQuote: QuoteResponse
+  tokenAddress: string
+}) {
+  return (
+    <View style={styles.amountContainer}>
+      {flow === CICOFlow.CashIn ? (
+        <TokenDisplay
+          amount={cicoQuote.quote.cryptoAmount}
+          tokenAddress={tokenAddress}
+          style={styles.amountText}
+          showLocalAmount={false}
+          testID="amount-crypto"
+        />
+      ) : (
+        <CurrencyDisplay
+          style={styles.amountText}
+          amount={{
+            value: cicoQuote.quote.cryptoAmount,
+            currencyCode: cicoQuote.quote.cryptoType,
+            localAmount: {
+              value: cicoQuote.quote.fiatAmount,
+              currencyCode: cicoQuote.quote.fiatType,
+              exchangeRate: 1,
+            },
+          }}
+          testID="amount-fiat"
+        />
+      )}
+    </View>
+  )
+}
+
+function TransactionDetails({
+  flow,
+  cicoQuote,
+  tokenAddress,
+}: {
+  flow: CICOFlow
+  cicoQuote: QuoteResponse
+  tokenAddress: string
+}) {
+  const { t } = useTranslation()
+  let tokenDisplay: string
+  switch (cicoQuote.quote.cryptoType) {
+    case CryptoType.cUSD:
+      tokenDisplay = t('celoDollar')
+      break
+    case CryptoType.cEUR:
+      tokenDisplay = t('celoEuro')
+      break
+    default:
+      tokenDisplay = t('total')
+  }
+  return (
+    <View style={styles.sectionContainer}>
+      <Text style={styles.sectionHeaderText}>
+        {t('fiatConnectReviewScreen.transactionDetails')}
+      </Text>
+      <LineItemRow
+        style={styles.sectionMainTextContainer}
+        textStyle={styles.sectionMainText}
+        title={
+          flow === CICOFlow.CashIn
+            ? t('fiatConnectReviewScreen.cashIn.transactionDetailsAmount')
+            : t('fiatConnectReviewScreen.cashOut.transactionDetailsAmount')
+        }
+        amount={
+          <CurrencyDisplay
+            amount={{
+              value: cicoQuote.quote.cryptoAmount,
+              currencyCode: cicoQuote.quote.cryptoType,
+              localAmount: {
+                value: cicoQuote.quote.fiatAmount,
+                currencyCode: cicoQuote.quote.fiatType,
+                exchangeRate: 1,
```

Comment:
Maybe it's something I don't understand about the component: how is exchangeRate of 1 determined?

Reply:
yeah this component is a little cumbersome. If `localAmount` is passed, no conversion happens and the `amount.value`, `amount.currencyCode` and `amount.localAmount.exchangeRate` are just ignored. Only `amount.localAmount.value` and `amount.localAmount.currencyCode` is used. I just passed 1 here since it's a required field. I'll add a comment to clarify this, but we need to clean up the `CurrencyDisplay` component